match/tests/Piipan.Match.Core.Tests/Parsers/OrchMatchRequestParserTests.cs

```diff
@@ -0,0 +1,113 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Piipan.Match.Api.Models;
+using Piipan.Match.Core.Parsers;
+using Moq;
+using Xunit;
+using FluentValidation;
+using FluentValidation.Results;
+
+namespace Piipan.Match.Core.Tests.Parsers
+{
+    public class OrchMatchRequestParserTests
+    {
+        [Fact]
+        public async Task EmptyStreamThrows()
+        {
+            // Arrange
+            var logger = Mock.Of<ILogger<OrchMatchRequestParser>>();
+            var validator = Mock.Of<IValidator<OrchMatchRequest>>();
+            var parser = new OrchMatchRequestParser(validator, logger);
+
+            // Act / Assert
+            await Assert.ThrowsAsync<StreamParserException>(() => parser.Parse(BuildStream("")));
+        }
+
+        [Theory]
+        [InlineData("{{")]
+        [InlineData("<xml>")]
+        [InlineData("{ data: 'foobar' }")]
+        [InlineData("{ data: []}")]
+        [InlineData("{ data: [{}]}")]
+        [InlineData("{ data: [{ssn: '000-00-0000'}]}")]
+        public async Task MalformedStreamThrows(string s)
+        {
+            // Arrange
+            var logger = Mock.Of<ILogger<OrchMatchRequestParser>>();
+            var validator = Mock.Of<IValidator<OrchMatchRequest>>();
+            var parser = new OrchMatchRequestParser(validator, logger);
+
+            // Act / Assert
+            await Assert.ThrowsAsync<StreamParserException>(() => parser.Parse(BuildStream(s)));
+        }
+
+        [Theory]
+        [InlineData(@"{ data: [{ lds_hash: 'abc' }]}", 1)] // invalid hash, but valid request
+        [InlineData(@"{ data: [{ lds_hash: '' }]}", 1)] // empty hash, but valid request
+        [InlineData(@"{'data':[

```

Comment:
```suggestion
        // farrington,1931-10-13,000-12-3456
        [InlineData(@"{'data':[

```
*Nitpick*: A small thing, but just trying to add comments where we use actual hashes.